### PR TITLE
Add Keyboard layout setting to the Welcome dialog

### LINKED ISSUE #6863: Add keyboard layout setting to the welcome dialog
I was writing a basic explanation on how to install NVDA today, and was wondering if the keyboard layout shouldn't be part of the welcome dialog. This is a setting that impacts your experience a lot when misconfigured.

### PR DESCRIPTION
**Pull request by dkager**
For this addition I didn't rewrite the dialog to use guiHelper. As a consequence I'm not sure if it works out visually.
Fixes #6863.

### CODE REVIEW

**Comment by feerrenrut** [comment]
I had a look at this to check the visual layout. Currently there is no gap (horizontally) between the label and `wx.Choice`. You can add a spacer for this. See `guiHelper.py` line 118. Also the vertical alignment of the `choice` control is wrong. It needs to be centred vertically. I think that this would be a good opportunity to convert this dialog to use the guiHelper.

**Comment by dkager** [comment]
Thank you. I'll move to guiHelper as part of this PR then.

**Comment by dkager** [comment]
Just sprinkled some guiHelper on this dialog.

File: source/gui/__init__.py [hunk 1]
```diff
@@ -578,43 +580,58 @@ def __init__(self, parent):
...
+		kbdLabelText = _("&Keyboard layout:")
+		layouts = keyboardHandler.KeyboardInputGesture.LAYOUTS
+		self.kbdNames = sorted(layouts)
+		kbdChoices = [layouts[layout] for layout in self.kbdNames]
+		self.kbdList = sHelper.addLabeledControl(kbdLabelText, wx.Choice, choices=kbdChoices)
+		try:
+			index = self.kbdNames.index(config.conf["keyboard"]["keyboardLayout"])
+			self.kbdList.SetSelection(index)
+		except:
+			log.debugWarning("Could not set Keyboard layout list to current layout",exc_info=True)
```
Comment (feerrenrut) [inline source/gui/__init__.py]: `log.debugWarning` does not produce a sound in NVDA releases. If this is hit I would suggest that it is a serious enough error that users of a released version of NVDA should be notified. I would propose changing this to `log.error`

**Review by feerrenrut — APPROVED** [review]
Looks good. I have built this locally to visually inspect and ensure the spacing and alignment is ok.

File: source/gui/__init__.py (hunk 1 above)
Comment (feerrenrut) [inline source/gui/__init__.py]: Updated my comment, `debugWarning` never produces a sound. `error` sounds are purposefully omitted from the release.

**Comment by dkager** [comment]
I copied the keyboard code from elsewhere. Arguably the welcome experience should be errorless from the user's perspective, even if under the hood everything's on fire. :)

**Comment by feerrenrut** [comment]
Understood, this will be the case in a release. However pre-release we want to know about problems as early as possible to have a chance to fix them prior to the release and RC versions.

**Comment by dkager** [comment]
Do you suppose this also goes for the same log message when setting the layout from the settings dialog? That's where the code came from.

**Comment by dkager** [comment]
Made it an error for the welcome dialog. Didn't touch the settings dialog though.

**Pull request merged**